packages/office-ui-fabric-react/src/components/ColorPicker/ColorPicker.base.tsx

```diff
@@ -164,40 +170,91 @@ export class ColorPickerBase extends BaseComponent<IColorPickerProps, IColorPick
   private _onTextChange(component: keyof IRGBHex, event: React.FormEvent<HTMLInputElement>, newValue?: string): void {
     const color = this.state.color;
     const isHex = component === 'hex';
-    if (String(color[component]) === newValue) {
+    // Trim values that are too long
+    newValue = (newValue || '').substr(0, isHex ? 6 : 3);
```

Comment:
Could there be constants for 6 and 3 (especially since 6 is also used on line 193)?